---
title: "R Script - Galapagos Penguin Gut Microbiome"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

__Rohrer SD, Jiménez-Uzcátegui G, Parker PG, & Chubiz LM. Composition and function of the Galapagos penguin gut microbiome vary with age, location, and a putative bacterial pathogen.__

## Import 16S Data

Read in 16S "shared" and "taxonomy" files generated by Mothur with corresponding metadata excel file using phyloseq and readxl. Convert metadata to phyloseq format. 
```{r eval=FALSE}
library(readxl)
library(tidyverse)
library(phyloseq)

mothur <- import_mothur(mothur_shared_file = "~/Research/Data/16S_Penguins_2018/R_penguins/GAPE2018.shared", mothur_constaxonomy_file = "~/Research/Data/16S_Penguins_2018/R_penguins/GAPE2018.taxonomy")
metadata_16s <- read_excel(path="~/Research/Data/16S_Penguins_2018/R_penguins/GAPE_metadata.xlsx")
meta16s <- sample_data(metadata_16s) 
rownames(meta16s) <-meta16s$id 
```
Merge files into one phyloseq object. Rename ranks and remove unclassified bacterial reads. Subsample to 11,295 reads (which corresponds to the sample with the lowest read depth greater than 10,000). 
```{r eval=FALSE}
gape <- merge_phyloseq(meta16s, mothur)
colnames(tax_table(gape)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus" )
sub <- subset_taxa(gape, Phylum != "Bacteria_unclassified")
sub <- rarefy_even_depth(sub, sample.size = 11295,
                         rngseed = TRUE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)
```
## View and plot top phyla and families

Now that the samples are combined and rarefied, turn the phyloseq object into a data frame and add a column for relative abundance based on count data. Arrange phyla based on relative abundance. 
```{r, eval=FALSE}
penguins_df <- psmelt(sub) 
penguins_df <- mutate(penguins_df, rel_abund=Abundance/11295)
peng_agg_phylum_data <- group_by(penguins_df, Sample, Phylum) %>%
  summarize(agg_rel_abund=sum(rel_abund)) %>%
  ungroup()
peng_agg_phylum_data %>%
  group_by(Phylum) %>%
  summarize(mean=mean(agg_rel_abund)) %>%
  arrange((desc(mean)))
```
Separate all phyla with relative abundances greater than 0.05%, set order of phyla and samples for the plot, and plot the abundances of these top phyla using ggplot. 
```{r eval=FALSE}
peng_top_phyla <- peng_agg_phylum_data %>%
  group_by(Phylum) %>%
  summarize(mean=mean(agg_rel_abund)) %>%
  arrange((desc(mean))) %>%
  top_n(6, mean) %>%
  pull(Phylum)
peng_top_phyla
## [1] "Fusobacteria" "Epsilonbacteraeota" "Firmicutes" "Proteobacteria" "Actinobacteria" "Bacteroidetes"     

df0 <- penguins_df %>% mutate(Phylum = fct_other(Phylum, peng_top_phyla))
df0$Phylum <- factor(df0$Phylum, levels = c("Fusobacteria", "Epsilonbacteraeota", "Firmicutes",
                                            "Proteobacteria", "Actinobacteria", "Bacteroidetes",
                                            "Other"))
df0$id <- factor(df0$id, levels = c("GAPE1808", "GAPE1809", "GAPE1812", "GAPE1832", "GAPE1833",
                                    "GAPE1834", "GAPE1838", "GAPE1841", "GAPE1852", "GAPE1853",
                                    "GAPE1860", "GAPE1861", "GAPE1862",  "GAPE1870", "GAPE1873",
                                    "GAPE1874", "GAPE1876", "GAPE1877", "GAPE1886","GAPE1893",
                                    "GAPE1895", "GAPE1898", "GAPE18101", "GAPE18103",
                                    "GAPE18105", "GAPE18106", "GAPE18107", "GAPE18115",
                                    "GAPE18119", "GAPE18120", "GAPE18121", "GAPE18122",
                                    "GAPE18124", "GAPE18128"))

phyla_plot <- ggplot(df0, aes(x=id, y=Abundance, fill=Phylum)) + 
  geom_bar(aes(), stat="identity", position="fill") +
  scale_fill_brewer(palette = "Paired", 
                    name = "Phyla", 
                    labels = c(expression(italic("Fusobacteria")), 
                               expression(italic("Epsilonbacteraeota")), 
                               expression(italic("Firmicutes")), 
                               expression(italic("Proteobacteria")), 
                               expression(italic("Actinobacteria")), 
                               expression(italic("Bacteroidetes")), 
                               "Other")) + 
  guides(fill=guide_legend(ncol=1)) + 
  ylab("Relative Abundance") + 
  theme_minimal()
phyla_plot <- phyla_plot + theme(axis.text.x = element_text(angle = -90, vjust = 0.5, hjust=1))
phyla_plot <- phyla_plot + theme(axis.title.x=element_blank())
phyla_plot <- phyla_plot + theme(legend.text.align = 0)
phyla_plot <- phyla_plot + theme(text = element_text(family = "serif"))
phyla_plot
```

```{r echo=FALSE, eval=FALSE }
ggsave("phyla_plot.png", width = 10, height = 5, dpi = 300)
```

```{r echo=FALSE, out.width = '100%'}
knitr::include_graphics("phyla_plot.png")
```

Repeat the same steps to sort and plot bacterial families with relative abundances >0.05%. 
```{r eval=FALSE}
peng_agg_family_data <- group_by(penguins_df, Sample, Family) %>%
  summarize(agg_rel_abund=sum(rel_abund)) %>%
  ungroup()
peng_agg_family_data %>%
  group_by(Family) %>%
  summarize(mean=mean(agg_rel_abund)) %>%
  arrange((desc(mean)))

peng_top_families <- peng_agg_family_data %>%
  group_by(Family) %>%
  summarize(mean=mean(agg_rel_abund)) %>%
  arrange((desc(mean))) %>%
  top_n(8, mean) %>%
  pull(Family)
peng_top_families
## [1] "Fusobacteriaceae" "Helicobacteraceae" "Clostridiaceae_1" "Pasteurellaceae"                 
## [5] "Peptostreptococcaceae" "Vibrionaceae" "Gammaproteobacteria_unclassified" "Lactobacillaceae"  

df1 <- penguins_df %>% mutate(Family = fct_other(Family, peng_top_families))

df1$Family <- factor(df1$Family, levels = c("Fusobacteriaceae", "Helicobacteraceae",
                                            "Clostridiaceae_1", "Pasteurellaceae",
                                            "Peptostreptococcaceae", "Vibrionaceae",
                                            "Gammaproteobacteria_unclassified",
                                            "Lactobacillaceae", "Other"), 
                     labels = c("Fusobacteriaceae", "Helicobacteraceae", "Clostridiaceae",
                                "Pasteurellaceae", "Peptostreptococcaceae", "Vibrionaceae",
                                "Unclassified Gammaproteobacteria", "Lactobacillaceae", "Other"))
df1$id <- factor(df1$id, levels = c("GAPE1808", "GAPE1809", "GAPE1812", "GAPE1832", "GAPE1833",
                                    "GAPE1834", "GAPE1838", "GAPE1841", "GAPE1852", "GAPE1853",
                                    "GAPE1860", "GAPE1861", "GAPE1862", "GAPE1870", "GAPE1873",
                                    "GAPE1874", "GAPE1876", "GAPE1877", "GAPE1886", "GAPE1893",
                                    "GAPE1895", "GAPE1898", "GAPE18101", "GAPE18103", "GAPE18105",
                                    "GAPE18106", "GAPE18107", "GAPE18115", "GAPE18119",
                                    "GAPE18120", "GAPE18121", "GAPE18122", "GAPE18124",
                                    "GAPE18128"))
gamma <- expression(paste("Unclassified ", italic("Gammaproteobacteria")))

family_plot <- ggplot(df1, aes(x=id, y=Abundance, fill=Family)) + 
  geom_bar(aes(), stat="identity", position="fill") +
  scale_fill_brewer(palette = "Paired", 
                    name = "Families", 
                    labels = c(expression(italic("Fusobacteriaceae")), 
                               expression(italic("Helicobacteraceae")), 
                               expression(italic("Pasteurellaceae")),
                               expression(italic("Clostridiaceae")), 
                               expression(italic("Peptostreptococcaceae")), 
                               expression(italic("Vibrionaceae")), 
                               gamma, 
                               expression(italic("Lactobacillaceae")),
                               "Other")) + 
  guides(fill=guide_legend(ncol=1)) + 
  ylab("Relative Abundance") + 
  theme_minimal()
family_plot <- family_plot + theme(axis.text.x = element_text(angle = -90, vjust = 0.5, hjust=1))
family_plot <- family_plot + theme(axis.title.x=element_blank())
family_plot <- family_plot + theme(legend.text.align = 0)
family_plot <- family_plot + theme(text = element_text(family = "serif"))
family_plot
```
```{r echo=FALSE, eval=FALSE }
ggsave("family_plot.png", width = 10, height = 5, dpi = 300)
```

```{r echo=FALSE, out.width = '100%'}
knitr::include_graphics("family_plot.png")
```

## Alpha Diversity
Next is to look at alpha diversity in the 16S dataset. Alpha diversity values have already been added to the metadata file, but here is how they were calculated from the subsampled phyloseq object: 
```{r eval=FALSE}
library(phyloseq)
erich <- estimate_richness(sub, measures=c("Observed", "Shannon", "Simpson"))
```
Use Kruskal Wallis tests to check whether alpha diversity significantly differs based on age using the metadata file. 
```{r eval=FALSE}
kruskal.test(observed ~ age, data = metadata_16s)

## Kruskal-Wallis rank sum test
## data:  observed by age
## Kruskal-Wallis chi-squared = 7.3238, df = 1, p-value = 0.006805

kruskal.test(shannon ~ age, data = metadata_16s)

## Kruskal-Wallis rank sum test
## data:  shannon by age
## Kruskal-Wallis chi-squared = 7.26, df = 1, p-value = 0.007051

kruskal.test(simpson ~ age, data = metadata_16s)

## Kruskal-Wallis rank sum test
## data:  simpson by age
## Kruskal-Wallis chi-squared = 8.0339, df = 1, p-value = 0.004591
```
Juveniles significantly differ from adults. Now remove rows corresponding to juvenile penguins and test alpha diversity significance between males and females (morphological sexing is not always possible for juvenile penguins and so this dataset includes several juveniles of unknown sex). 
```{r eval=FALSE}
adultsalpha <- subset(metadata_16s, age=="Adult")
adultsalpha
kruskal.test(observed ~ sex, data = adultsalpha)

## Kruskal-Wallis rank sum test
## data:  observed by sex
## Kruskal-Wallis chi-squared = 1.7646, df = 1, p-value = 0.1841

kruskal.test(shannon ~ sex, data = adultsalpha)

## Kruskal-Wallis rank sum test
## data:  shannon by sex
## Kruskal-Wallis chi-squared = 0.022857, df = 1, p-value = 0.8798

kruskal.test(simpson ~ sex, data = adultsalpha)

## Kruskal-Wallis rank sum test
## data:  simpson by sex
## Kruskal-Wallis chi-squared = 0.051429, df = 1, p-value = 0.8206
```
No significant difference is apparent based on sex. Now test alpha diversity based on location, first excluding a single sample from the northern site El Muneco. One other sample was from Playa Perros, a site only a few miles from Puerto Pajas - including this sample grouped with Puerto Pajas or excluding it completely did not change significance for any test. 
```{r eval=FALSE}
mainsites <- subset(metadata_16s, location!="el_muneco")
mainsites <- subset(mainsites,location!="playa_perros")
kruskal.test(observed ~ location, data = mainsites)

## Kruskal-Wallis rank sum test
## data:  observed by location
## Kruskal-Wallis chi-squared = 1.6641, df = 2, p-value = 0.4352

kruskal.test(shannon ~ location, data = mainsites)

## Kruskal-Wallis rank sum test
## data:  shannon by location
## Kruskal-Wallis chi-squared = 6.5108, df = 2, p-value = 0.03857

kruskal.test(simpson ~ location, data = mainsites)

## Kruskal-Wallis rank sum test
## data:  simpson by location
## Kruskal-Wallis chi-squared = 5.1039, df = 2, p-value = 0.07793
```
Shannon diversity does significantly differ based on location when limited to the three main sites (Caleta Iguana, Puerto Pajas, and Marielas); however, observed richness and simpson's diversity do not significantly vary. 

### Alpha Diversity Plots
Plot observed, shannon, and simpson diversity across age classes. 
```{r eval=FALSE}
library(cowplot)
library(ggpubr)

obs <- ggboxplot(metadata_16s, x = "age", y = "observed", color = "age", 
                 add = "jitter", legend = "none") +
                scale_color_manual(name=NULL, 
                     values = c("darkgreen", "blue"), 
                     breaks = c("Adult", "Juvenile")) + 
  labs(title=NULL, 
       X=NULL, 
       y="Observed Richness") + 
  stat_compare_means(method = "kruskal.test", label = "p.format", fontface = 'bold')
observed <- obs + theme(axis.title.x = element_blank())
observed <- observed + theme(text = element_text(family = "serif"))

shan <- ggboxplot(metadata_16s, x = "age", y = "shannon", color = "age", 
                  add = "jitter", legend = "none") + 
  scale_color_manual(name=NULL, 
                     values=c("darkgreen", "blue"), 
                     breaks=c("Adult", "Juvenile")) + 
  labs(title=NULL, 
       X=NULL, 
       y="Shannon Diversity Index", fontface = 'bold') + 
  stat_compare_means(method = "kruskal.test", label = "p.format", fontface = 'bold') 
shannon <- shan + theme(axis.title.x = element_blank())
shannon <- shannon + theme(text = element_text(family = "serif"))

simp <- ggboxplot(metadata_16s, x = "age", y = "simpson", color = "age", 
                  add = "jitter", legend = "none") + 
  scale_color_manual(name=NULL, 
                     values=c("darkgreen", "blue"), 
                     breaks=c("Adult", "Juvenile")) + 
  labs(title=NULL, 
       X=NULL, 
       y="Simpson's Diversity Index") + 
  stat_compare_means(method = "kruskal.test", label = "p.format", fontface = 'bold')
simpson <- simp + theme(axis.title.x = element_blank())
simpson <- simpson + theme(text = element_text(family = "serif"))
simpson
                           
#combine
theme_set(theme_cowplot(font_family = "serif") + 
          theme(text = element_text(colour = "black")))
row <- plot_grid(observed, simpson, shannon, ncol = 3, labels = "AUTO")
plot_grid(row, nrow = 1) + 
  theme_bw()
```
```{r echo=FALSE, eval=FALSE }
ggsave("alpha_age.png", width = 9, height = 3, dpi = 300)
```

```{r echo=FALSE, out.width = '100%'}
knitr::include_graphics("alpha_age.png")
```
Plot observed, shannon, and simpson diversity across three primary locations. Verify that the Kruskal Wallis results generated by ggpubr in these plots match the values obtained earlier. 
```{r eval=FALSE}
library(cowplot)
library(ggpubr)

obs_sites <- ggboxplot(mainsites, x = "site_abbr", y = "observed", color = "sitename", 
                       add = "jitter", legend = "none") + 
  scale_color_manual(name=NULL, 
                     values=c("blue", "darkorange", "darkgreen"), 
                     breaks=c("Caleta Iguana", "Puerto Pajas", "Marielas")) + 
  labs(title=NULL, 
       X=NULL, 
       y="Observed Richness") + 
stat_compare_means(method = "kruskal.test", label = "p.format", fontface = 'bold')
observed_sites <- obs_sites + theme(axis.title.x = element_blank())
observed_sites <- observed_sites + theme(text = element_text(family = "serif"))

shan_sites <- ggboxplot(mainsites, x = "site_abbr", y = "shannon", color = "sitename", 
                        add = "jitter", legend = "none") + 
  scale_color_manual(name=NULL, 
                     values=c("blue", "darkorange", "darkgreen"), 
                     breaks=c("Caleta Iguana", "Puerto Pajas", "Marielas")) + 
  labs(title=NULL, 
       X=NULL, 
       y="Shannon Diversity Index", fontface = 'bold') +
stat_compare_means(method = "kruskal.test", label = "p.format", fontface = 'bold')
shannon_sites <- shan_sites + theme(axis.title.x = element_blank())
shannon_sites <- shannon_sites + theme(text = element_text(family = "serif"))

simp_sites <- ggboxplot(mainsites, x = "site_abbr", y = "simpson", color = "sitename", 
                        add = "jitter", legend = "none") + 
  scale_color_manual(name=NULL, 
                     values=c("blue", "darkorange", "darkgreen"), 
                     breaks=c("Caleta Iguana", "Puerto Pajas", "Marielas")) + 
  labs(title=NULL, 
       X=NULL, 
       y="Simpson's Diversity Index")  +
  stat_compare_means(method = "kruskal.test", label = "p.format", fontface = 'bold')
simpson_sites <- simp_sites + theme(axis.title.x = element_blank())
simpson_sites <- simpson_sites + theme(text = element_text(family = "serif"))

#combine
row <- plot_grid(observed_sites, simpson_sites, shannon_sites, ncol = 3, labels = "AUTO")
plot_grid(row, nrow = 1)
```
## Beta Diversity
Use PERMANOVAs in the package vegan to see if age, sex, or location significantly differ. Check dispersion with PERMDISP. Must control for location in the model to effectively test age-based variation. First remove single samples from El Muneco and Playa Perros and four samples from Caleta Iguana to only include samples from the two sites with the most samples, since PERMANOVAs can be sensitive to unbalanced designs. 

```{r eval=FALSE}
library(vegan)
site_ps <- subset_samples(sub, location!="el_muneco")
site_ps <- subset_samples(site_ps, location!="playa_perros")

site_pm <- subset_samples(site_ps, location!="caleta_iguana")
```
Now calculate jaccard distances between samples and make a data frame of sample data. 
```{r eval=FALSE}
sitepm_jac <- phyloseq::distance(site_pm, method = "jaccard")
sitepmdf <- data.frame(sample_data(site_pm))
```
Test age, sex, and location using three different models: one with locations set as blocks (strata) and the variables listed as "age + sex", one with locations set as blocks (strata) and the variables listed as "sex + age", and one with location listed as another variable in the model. 
```{r eval=FALSE}
perm <- how(nperm = 999)
setBlocks(perm) <- with(sitepmdf, location)
adonis2(sitepm_jac ~ age + sex, data = sitepmdf, permutations = perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(sitepmdf, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = sitepm_jac ~ age + sex, data = sitepmdf, permutations = perm)
##          Df SumOfSqs      R2      F Pr(>F)  
## age       1   0.6376 0.08833 2.4878  0.013 *
## sex       2   0.4298 0.05954 0.8384  0.616  
## Residual 24   6.1510 0.85213                
## Total    27   7.2184 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(sitepm_jac ~ sex + age, data = sitepmdf, permutations = perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(sitepmdf, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = sitepm_jac ~ sex + age, data = sitepmdf, permutations = perm)
##          Df SumOfSqs      R2      F Pr(>F)  
## sex       2   0.4533 0.06280 0.8843  0.535  
## age       1   0.6140 0.08507 2.3959  0.016 *
## Residual 24   6.1510 0.85213                
## Total    27   7.2184 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(sitepm_jac ~ age + sex + location, data = sitepmdf)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = sitepm_jac ~ age + sex + location, data = sitepmdf)
##          Df SumOfSqs      R2      F Pr(>F)  
## age       1   0.6376 0.08833 2.5839  0.013 *
## sex       2   0.4298 0.05954 0.8708  0.598  
## location  1   0.4756 0.06588 1.9273  0.061 .
## Residual 23   5.6754 0.78625                
## Total    27   7.2184 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```
In all cases, age classes significantly vary. However, dispersion must also be tested to ensure the age result is not being confounded by significant dispersion. Sex does not significantly differ, and the two locations (Puerto Pajas and Marielas) do not significantly differ. 
```{r eval=FALSE}
beta_age <- betadisper(sitepm_jac, sitepmdf$age)
beta_age
permutest(beta_age)

## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999

## Response: Distances
##           Df  Sum Sq  Mean Sq      F N.Perm Pr(>F)
## Groups     1 0.02804 0.028045 1.7929    999  0.194
## Residuals 26 0.40670 0.015643    

```
PERMDISP is not significant, so the age effect appears valid. Check one other way by adding Caleta Iguana as an additional level to make sure the age and location patterns are similar across sites, and finally by including all sites including the two single-sample locations to see if the age result persists. 
```{r eval=FALSE}
site_ps <- subset_samples(sub, location!="el_muneco")
site_ps <- subset_samples(site_ps, location!="playa_perros")
siteps_jac <- phyloseq::distance(site_ps, method = "jaccard")
sitepsdf <- data.frame(sample_data(site_ps))

perm <- how(nperm = 999)
setBlocks(perm) <- with(sitepsdf, location)
adonis2(siteps_jac ~ age + sex, data = sitepsdf, permutations = perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(sitepsdf, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = siteps_jac ~ age + sex, data = sitepsdf, permutations = perm)
##          Df SumOfSqs      R2      F Pr(>F)  
## age       1   0.5833 0.07091 2.2677  0.021 *
## sex       2   0.4400 0.05349 0.8553  0.607  
## Residual 28   7.2027 0.87559                
## Total    31   8.2261 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(siteps_jac ~ age + sex + location, data = sitepsdf)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = siteps_jac ~ age + sex + location, data = sitepsdf)
##          Df SumOfSqs      R2      F Pr(>F)  
## age       1   0.5833 0.07091 2.3734  0.023 *
## sex       2   0.4400 0.05349 0.8952  0.592  
## location  2   0.8124 0.09875 1.6526  0.058 .
## Residual 26   6.3903 0.77684                
## Total    31   8.2261 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

beta_age2 <- betadisper(siteps_jac, sitepsdf$age)
beta_age2
permutest(beta_age2)

## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999

## Response: Distances
##           Df  Sum Sq   Mean Sq      F N.Perm Pr(>F)
## Groups     1 0.00935 0.0093497 0.6779    999  0.399
## Residuals 30 0.41375 0.0137916  

#Try without excluding any samples
sub_jac <- phyloseq::distance(sub, method = "jaccard")
subdf <- data.frame(sample_data(sub))

## adonis2(sub_jac ~ location + age, data = subdf)
## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = sub_jac ~ location + age, data = subdf)
##          Df SumOfSqs      R2      F Pr(>F)  
## location  4   1.3083 0.14933 1.3324  0.068 .
## age       1   0.5796 0.06615 2.3610  0.016 *
## Residual 28   6.8733 0.78452                
## Total    33   8.7611 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

beta_age3 <- betadisper(sub_jac, subdf$age)
beta_age3
permutest(beta_age3)

## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999

## Response: Distances
##           Df  Sum Sq   Mean Sq      F N.Perm Pr(>F)
## Groups     1 0.00072 0.0007183 0.0495    999  0.816
## Residuals 32 0.46399 0.0144996  

```
Even with sparsely sampled locations included, the P value remains similar for juveniles vs. adults when location is accounted for in the model. These models included samples of unknown sex since all juveniles were included, so verify the nonsignificant result for sex by excluding juveniles and restricting model to males vs. females in adults. 
```{r eval=FALSE}
adults <- subset_samples(sub, age!="Juvenile")
adults <- subset_samples(adults, location!="el_muneco")
adults

adults_jac <- phyloseq::distance(adults, method = "jaccard")
adultsdf <- data.frame(sample_data(adults))

perm <- how(nperm = 999)
setBlocks(perm) <- with(adultsdf, location)
adonis2(adults_jac ~ sex, data = adultsdf, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(adultsdf, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = adults_jac ~ sex, data = adultsdf, permutations = perm)
##          Df SumOfSqs      R2      F Pr(>F)
## sex       1   0.2799 0.05771 1.1023  0.233
## Residual 18   4.5707 0.94229              
## Total    19   4.8506 1.00000   
```
Males and females do not significantly differ among adults. The first model showed that Puerto Pajas and Marielas do not have significantly different beta diversity, but now include the Caleta Iguana samples as well as the sample from Playa Perros (very near Puerto Pajas) and look at the sheltered site (Marielas) vs. the more open sites. 
```{r eval=FALSE}
site_ps <- subset_samples(sub, location!="el_muneco")
site_jac <- phyloseq::distance(site_ps, method = "jaccard")
sitedf <- data.frame(sample_data(site_ps))

adonis2(site_jac ~ age + site_exposed, data = sitedf)
## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = site_jac ~ age + site_exposed, data = sitedf)
##              Df SumOfSqs      R2      F Pr(>F)  
## age           1   0.5714 0.06828 2.3489  0.023 *
## site_exposed  1   0.4996 0.05970 2.0538  0.038 *
## Residual     30   7.2978 0.87202                
## Total        32   8.3689 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(site_jac ~ site_exposed + age, data = sitedf)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = site_jac ~ site_exposed + age, data = sitedf)
##              Df SumOfSqs      R2      F Pr(>F)  
## site_exposed  1   0.5188 0.06199 2.1325  0.035 *
## age           1   0.5523 0.06599 2.2702  0.021 *
## Residual     30   7.2978 0.87202                
## Total        32   8.3689 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

beta <- betadisper(site_jac, sitedf$site_exposed)
permutest(beta)

## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999

## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)
## Groups     1 0.007127 0.0071274 1.1008    999  0.293
## Residuals 31 0.200727 0.0064751  
```
Marielas is significantly different from the more exposed sites, with and without removing effects of age first. PERMDISP is not significant, meaning the result is not confounded by dispersion. Now ordinate phyloseq object using jaccard distance to plot the three main sites: Caleta Iguana, Puerto Pajas, and Marielas. 
```{r eval=FALSE}
library(phyloseq)

site_ps <- subset_samples(sub, location!="el_muneco")
site_ps <- subset_samples(site_ps, location!="playa_perros")
location_pcoa <- ordinate(
  physeq = site_ps, 
  method = "PCoA", 
  distance = "jaccard"
)
```
Plot PCoA using phyloseq and ggplot. 
```{r eval=FALSE}
pcoa_title <- expression(italic(paste("Principal Coordinates Analysis")))

pcoa_site <- plot_ordination(
  physeq = site_ps,
  ordination = location_pcoa,
  color = "location",
  title = pcoa_title
) + 
  scale_color_manual(name=NULL,
                     values=c("blue", "darkorange", "darkgreen"),
                     breaks=c("caleta_iguana", "puerto_pajas", "marielas"),
                     labels=c("CI", "PP", "M")
  ) +
  geom_point(size = 2.5) + 
  coord_cartesian() + 
  stat_ellipse() + 
  theme_light()
pcoa_site <- pcoa_site + theme(text = element_text(family = "serif"))
pcoa_site
```
Now add a map of sample sites using stamen maps with ggmap() and plot alpha and beta diversity together in one plot. 
```{r eval=FALSE}
library(ggmap) 
library(ggsn)
library(ggplot2)
library(ggrepel)
library(readxl)
library(grid)

sites <- read_excel(path="~/Research/Data/WGS/R_WGS/Coordinates.xlsx")
islands <- read_excel(path="~/Research/Data/WGS/R_WGS/islands2.xlsx")
volcano <- read_excel(path="~/Research/Data/WGS/R_WGS/volcano.xlsx")

#Make map of Galapagos Islands
gpscoords <- c(-91.75, -1.47, -89.15, 0.71)
gps <- get_map(location = gpscoords, source = "stamen", maptype = "terrain", zoom = 11)

#Box around Isabela Island
bbx <- c(left=-91.7,bottom=-1.15,right=-90.65,top=0.2)
x <- c(bbx["left"], bbx["left"], bbx["right"], bbx["right"])
y <- c(bbx["bottom"], bbx["top"], bbx["top"], bbx["bottom"])
df <- data.frame(x, y)
names(df) <- c("long", "lat")

#Combine Galapagos map with box indicating Isabela
big_map <- ggmap(gps) + 
  labs(title = NULL) + 
  geom_polygon(aes(x=x, y=y), data=df, fill=NA, col = "red", size = 1.05) + 
    theme(panel.border = element_rect(fill = NA, size = 1.4),
        axis.title = element_blank(), 
        axis.text = element_blank(), 
        axis.ticks = element_blank())
big_map <- big_map + theme(plot.background = element_blank())

#Get map of Isabela Island
isacoords <- c(-91.7, -1.15, -90.65, 0.2)
isabela <- get_map(location = isacoords, source = "stamen", maptype = "terrain", zoom = 11)

#Plot Isabela Island map with scalebar 
isa_map <- ggmap(isabela) + 
  xlab("Longitude") + 
  ylab("Latitude") + 
  geom_point(data = sites, aes(x = Longitude, y = Latitude, color=Site), size = 3, family = "serif") +
  scale_color_manual(name=NULL, 
                  values=c("blue", "orange", "darkgreen"), 
                  breaks=c("Caleta Iguana", "Puerto Pajas", "Marielas")) + 
  geom_text_repel(data=sites, aes(x = Longitude, y = Latitude, label = Site, color=Site), 
                  fontface = "bold", color = "gray8", size = 3.1, family = "serif") + 
  geom_text(data=islands, aes(x = Longitude, y = Latitude, label = Island), 
                  fontface = "italic", color = "gray10", size = 4.5, family = "serif") + 
  geom_text(data=volcano, aes(x = Longitude, y = Latitude, label = Volcano), 
                  fontface = "italic", color = "gray14", size = 3, family = "serif") + 
  scalebar(data = df, dist = 15, dist_unit = "km", st.bottom = FALSE, 
                  transform = TRUE, st.dist = 0.025, st.size = 3.2, model = "Airy", 
                  location = "bottomleft", anchor = c(x = -91.686, y = -1.14)) + 
  theme_bw()
isa_map <- isa_map + theme(legend.position = "none")
isa_map <- isa_map + theme(text = element_text(family = "serif"))
isa_map

#Add north symbol
library(patchwork)
library(png)

n <- ("north.png")
npng <- readPNG(n, native=TRUE)
isa <- isa_map + 
  inset_element(p = npng, left = 0.01, bottom = 0.87, right = 0.19, top = 0.99) + 
  theme_void()

#Combine maps
all_map <- isa + inset_element(ggplotGrob(big_map), 
                               left = 0.59, right = 1, bottom = 0.71, top = 1, 
                               align_to = 'full') + 
                  theme_void()
all_map

##cowplot
row <- plot_grid(observed_sites, simpson_sites, shannon_sites, ncol = 3, labels = "AUTO")
row2 <- plot_grid(pcoa_site, all_map, ncol = 2, labels = c('D', 'E'), rel_widths = c(1.4, 1))
plot_grid(NULL, row, NULL, row2, nrow = 4, rel_heights = c(0.03, 0.55, 0.03, 0.8)) + 
  theme_bw()

```


```{r echo=FALSE, eval=FALSE }
ggsave("cowplot_location.png", width = 10, height = 7.3, dpi = 300)

```
```{r echo=FALSE, out.width = '100%'}
knitr::include_graphics("cowplot_location.png")

```

## Metabolic KEGG Pathways

Import metabolic KEGG pathway data generated by MetaErg contig annotation. KEGG results have been saved in formats compatible with conversion to a phyloseq object (i.e. a name file, count file, and metadata). 
```{r eval=FALSE}
kegg_count <- read_excel("~/Research/Data/WGS/R_WGS/METAB_count.xlsx")
kegg_names <- read_excel("~/Research/Data/WGS/R_WGS/METAB_names.xlsx")
kegg_metadata <- read_excel(path="~/Research/Data/WGS/R_WGS/WGS_metadata.xlsx")
meta_kegg <- sample_data(kegg_metadata) 
rownames(meta_kegg) <-meta_kegg$id 

kegg_count <- column_to_rownames(kegg_count, var = "OTUID")
kegg_names <- column_to_rownames(kegg_names, var = "OTUID")
kegg_names <- as.matrix(kegg_names)

COUNT = otu_table(kegg_count, taxa_are_rows=TRUE)
TAX = tax_table(kegg_names)
kegg_physeq <- phyloseq(COUNT, TAX)
kegg <- merge_phyloseq(meta_kegg, kegg_physeq)
```
Check metabolic pathway counts, then subsample to 19,684 (excludes two sparse samples). Convert phyloseq object into a data frame. 
```{r eval=FALSE}
sample_sums(kegg)
kegg_sub <- rarefy_even_depth(kegg, sample.size = 19684, rngseed = TRUE, replace = TRUE, verbose =
                                TRUE)

kegg_df <- psmelt(kegg_sub)
```
Plot relative abundances of KEGG metabolic pathways using ggplot. 
```{r eval=FALSE}
kegg_df$Sample <- factor(kegg_df$Sample, levels = c("GAPE1809", "GAPE1812", "GAPE1833",
                                                    "GAPE1834", "GAPE1838", "GAPE1852",
                                                    "GAPE1861", "GAPE1862", "GAPE1870",
                                                    "GAPE1886", "GAPE1893", "GAPE18101",
                                                    "GAPE18105", "GAPE18106", "GAPE18107",
                                                    "GAPE18115", "GAPE18120", "GAPE18122"))
                                   
kegg_plot <- ggplot(kegg_df, aes(x=Sample, y=Abundance, fill=Class)) + 
  geom_bar(aes(), stat="identity", position="fill") +
  scale_fill_brewer(palette = "Spectral") + 
  guides(fill=guide_legend(ncol=1)) + 
  ylab("Relative Abundance") + 
  theme_minimal()
kegg_plot <- kegg_plot + theme(axis.text.x = element_text(angle = -90, vjust = 0.5, hjust=1))
kegg_plot <- kegg_plot + theme(axis.title.x=element_blank())
kegg_plot <- kegg_plot + guides(fill=guide_legend(title="Metabolic Groups"))
kegg_plot <- kegg_plot + theme(text = element_text(family = "serif"))
kegg_plot
```
Combine the phylum, family, and pathway graphs into one plot. 
```{r eval=FALSE}
library(ggpubr)
library(cowplot)
row <- plot_grid(phyla_plot, ncol = 1, labels = "AUTO")
row2 <- plot_grid(family_plot, ncol = 1, labels = "B")
row3 <- plot_grid(kegg_plot, ncol = 1, labels = "C")
plot_grid(row, row2, row3, nrow = 3, align="v", rel_heights = c(1, 1, 1)) + 
  theme_bw()
```

```{r echo=FALSE, eval=FALSE }
ggsave("cowplot_abundance.png", width = 10, height =11, dpi = 300)
```

```{r echo=FALSE, out.width = '100%'}
knitr::include_graphics("cowplot_abundance.png")
```

## _C. perfringens_
Use the metabolic pathway dataset that was imported in the previous section to check for community variation based on _C. perfringens_ virulence factor presence. First use PERMANOVA and PERMDISP to check for significant differences between groups. 
```{r eval=FALSE}
kegg_jac <- phyloseq::distance(kegg_sub, method = "jaccard")
kegg_df <- data.frame(sample_data(kegg_sub))

perm <- how(nperm = 999)
setBlocks(perm) <- with(kegg_df, location)

adonis2(kegg_jac ~ age + sex + Cperfringens_VF, data = kegg_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(kegg_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = kegg_jac ~ age + sex + Cperfringens_VF, data = kegg_df, permutations = perm)
##                 Df SumOfSqs      R2      F Pr(>F)  
## age              1  0.03135 0.07268 1.4636  0.176  
## sex              2  0.02465 0.05715 0.5755  0.880  
## Cperfringens_VF  1  0.09690 0.22465 4.5242  0.014 *
## Residual        13  0.27843 0.64552                
## Total           17  0.43133 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(kegg_jac ~  Cperfringens_VF + age + sex, data = kegg_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(kegg_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = kegg_jac ~ Cperfringens_VF + age + sex, data = kegg_df, permutations = perm)
##                 Df SumOfSqs      R2      F Pr(>F)   
## Cperfringens_VF  1  0.10733 0.24884 5.0114  0.006 **
## age              1  0.01911 0.04431 0.8924  0.419   
## sex              2  0.02645 0.06133 0.6175  0.855   
## Residual        13  0.27843 0.64552                 
## Total           17  0.43133 1.00000                 
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(kegg_jac ~ age*Cperfringens_VF, data = kegg_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(kegg_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = kegg_jac ~ age * Cperfringens_VF, data = kegg_df, permutations = perm)
##                     Df SumOfSqs      R2      F Pr(>F)   
## age                  1  0.03135 0.07268 1.5667  0.126   
## Cperfringens_VF      1  0.09510 0.22048 4.7528  0.006 **
## age:Cperfringens_VF  1  0.02476 0.05740 1.2373  0.309   
## Residual            14  0.28012 0.64945                 
## Total               17  0.43133 1.00000                 
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(kegg_jac ~ sex*Cperfringens_VF, data = kegg_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(kegg_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = kegg_jac ~ sex * Cperfringens_VF, data = kegg_df, permutations = perm)
##                     Df SumOfSqs      R2      F Pr(>F)   
## sex                  2  0.02562 0.05940 0.6267  0.841   
## Cperfringens_VF      1  0.10845 0.25144 5.3059  0.004 **
## sex:Cperfringens_VF  2  0.05198 0.12050 1.2714  0.367   
## Residual            12  0.24528 0.56866                 
## Total               17  0.43133 1.00000                 
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(kegg_jac ~  Cperfringens_VF + location, data = kegg_df)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = kegg_jac ~ Cperfringens_VF + location, data = kegg_df)
##                 Df SumOfSqs      R2      F Pr(>F)   
## Cperfringens_VF  1  0.10733 0.24884 5.3470  0.004 **

adonis2(kegg_jac ~  Cperfringens_VF*location, data = kegg_df)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = kegg_jac ~ Cperfringens_VF * location, data = kegg_df)
##                          Df SumOfSqs      R2      F Pr(>F)   
## Cperfringens_VF           1  0.10733 0.24884 5.6653  0.003 **
## location                  3  0.06304 0.14616 1.1092  0.353   
## Cperfringens_VF:location  2  0.05255 0.12183 1.3868  0.213   
## Residual                 11  0.20840 0.48317                 
## Total                    17  0.43133 1.00000                 
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

beta <- betadisper(kegg_jac, kegg_df$Cperfringens_VF)
permutest(beta)

## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999

## Response: Distances
##           Df    Sum Sq   Mean Sq      F N.Perm Pr(>F)  
## Groups     1 0.0044085 0.0044085 3.4198    999  0.081 .
## Residuals 16 0.0206254 0.0012891                       
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```
Groups with and without _C. perfringens_ virulence factors significantly differ in a PERMANOVA model when controlling for location, and PERMDISP is not significant between groups. No evidence for variation based on age, sex, or location using metabolic pathway data. No evidence for interactive effects between _C. perfringens_ with age, sex, or location. Now ordinate with jaccard distance using the package phyloseq. 
```{r eval=FALSE}
pcoa_metab <- ordinate(
  physeq = kegg_sub, 
  method = "PCoA", 
  distance = "jaccard"
)
```
Plot ordination of metabolic pathways based on _C. perfringens_ presence using the significance value from the first PERMANOVA. 
```{r eval=FALSE}
cperf_metab <- expression(italic(paste("Metabolic pathway clustering (P=0.014)")))

metab_pcoa <- plot_ordination(
  physeq = kegg_sub,
  ordination = pcoa_metab,
  color = "Cperfringens_VF", 
  title = cperf_metab
) + 
  scale_color_manual(name=NULL,
                     values=c("black", "red"),
                     breaks=c("N", "Y"),
                     labels=c("No VFs", "VFs")
  ) +
  geom_point(aes(color = Cperfringens_VF), alpha = 1, size = 2.5) +
  stat_ellipse() + 
  coord_cartesian() + 
  theme_light()
metab_pcoa <- metab_pcoa + theme(text = element_text(family = "serif"))
metab_pcoa
```
Use Kruskal Wallis tests with false discovery correction to determine significantly different metabolic pathways between samples with and without _C. perfringens_ virulence factors detected. First convert phyloseq object to a data frame and then add an extra column for relative abundance by dividing from subsampled total read count. 
```{r eval=FALSE}
df_kegg <- psmelt(kegg_sub)
df_kegg <- mutate(df_kegg, rel_abund=Abundance/19684)
```
Sort based on descending order in the relative abundance column. Separate all pathways with adjusted p values <0.05. 
```{r eval=FALSE}
library(broom)
library(purrr)
kegg_agg_class_data <- group_by(df_kegg, Sample, Class, Cperfringens_VF) %>%
  summarize(agg_rel_abund=sum(rel_abund)) %>%
  ungroup()

kegg_class_tests <- kegg_agg_class_data %>%
  nest(sample_data = c(-Class)) %>%
  mutate(test=map(sample_data, ~tidy(kruskal.test(agg_rel_abund~Cperfringens_VF, data=.)))) %>%
  unnest(test) %>%
  mutate(p.value.adj=p.adjust(p.value, method="BH")) %>%
  arrange(p.value.adj)
kegg_class_tests

kegg_sig_class <- kegg_class_tests %>%
  filter(p.value.adj <= 0.05) %>%
  pull(Class)
kegg_sig_class
## [1] "Amino Acid Metabolism" "Energy Metabolism"     "Lipid Metabolism"     
```
Amino acid metabolism, energy metabolism, and lipid metabolism significantly vary between groups. Plot these pathways on a boxplot. 
```{r eval=FALSE}
cperf_metab_groups <- expression(italic(paste("Metabolic groups differ (adjusted P<0.05)")))

metab_groups <- kegg_agg_class_data %>%
  filter(Class %in% kegg_sig_class) %>%
  mutate(Class=factor(Class, levels=kegg_sig_class)) %>%
  ggplot(aes(x=Class, y=agg_rel_abund, color=Cperfringens_VF)) +
  geom_boxplot() +
  scale_color_manual(name=NULL,
                     values=c("black", "red"),
                     breaks=c("N", "Y"),
                     labels=c("No VFs", "VFs")) +
  labs(title=cperf_metab_groups,
       x=NULL,
       y="Relative Abundance") +
  theme_classic()
metab_groups <- metab_groups + theme(axis.text.x = element_text(angle = 25, hjust = 1))
metab_groups <- metab_groups + theme(text = element_text(family = "serif"))
metab_groups
```
Now look at protein family data generated by MetaErg to see if variation is also evident between these groups there. First import the datasets and merge into one phyloseq object. 
```{r eval=FALSE}
pfamme_count <- read_excel("~/Research/Data/WGS/R_WGS/pfamme_count.xlsx")
pfamme_names <- read_excel("~/Research/Data/WGS/R_WGS/pfamme_names.xlsx")
metadata <- read_excel("~/Research/Data/WGS/R_WGS/WGS_metadata.xlsx")
pfamme_count <- tibble::as_tibble(pfamme_count)
pfamme_count <- column_to_rownames(pfamme_count, var = "OTU")
pfamme_names <- tibble::as_tibble(pfamme_names)
pfamme_names <- column_to_rownames(pfamme_names, var = "OTU")
metadata <- tibble::as_tibble(metadata)
metadata <- column_to_rownames(metadata, var = "id")
pfamme_names <- as.matrix(pfamme_names)
COUNT = otu_table(pfamme_count, taxa_are_rows=TRUE)
TAX = tax_table(pfamme_names)
map <- sample_data(metadata)
pfamme_ps <- phyloseq(COUNT, TAX)
pfamme <- merge_phyloseq(map, pfamme_ps)
pfamme
```
Subsample to the lowest reasonable read depth (excludes one sample). 
```{r eval=FALSE}
sample_sums(pfamme)
pfamme_sub <- rarefy_even_depth(pfamme, sample.size = 22259, rngseed = TRUE, replace = TRUE, verbose = TRUE)
sample_sums(pfamme_sub)
```
Check significance of age, sex, and _C. perfringens_ virulence factor presence, using PERMANOVA and PERMDISP. 
```{r eval=FALSE}
pfam_jac <- phyloseq::distance(pfamme, method = "jaccard")
pfam_df <- data.frame(sample_data(pfamme))

perm <- how(nperm = 999)
setBlocks(perm) <- with(pfam_df, location)

adonis2(pfam_jac ~ age + sex + Cperfringens_VF, data = pfam_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(pfam_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = pfam_jac ~ age + sex + Cperfringens_VF, data = pfam_df, permutations = perm)
##                 Df SumOfSqs      R2      F Pr(>F)  
## age              1   0.2518 0.05403 1.0779  0.342  
## sex              2   0.2446 0.05250 0.5236  0.961  
## Cperfringens_VF  1   0.6597 0.14156 2.8241  0.020 *
## Residual        15   3.5041 0.75191                
## Total           19   4.6603 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(pfam_jac ~  Cperfringens_VF + age + sex, data = pfam_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(pfam_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = pfam_jac ~ Cperfringens_VF + age + sex, data = pfam_df, permutations = perm)
##                 Df SumOfSqs      R2      F Pr(>F)  
## Cperfringens_VF  1   0.6200 0.13303 2.6538  0.028 *
## age              1   0.2916 0.06257 1.2482  0.283  
## sex              2   0.2446 0.05250 0.5236  0.965  
## Residual        15   3.5041 0.75191                
## Total           19   4.6603 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(pfam_jac ~ age*Cperfringens_VF, data = pfam_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(pfam_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = pfam_jac ~ age * Cperfringens_VF, data = pfam_df, permutations = perm)
##                     Df SumOfSqs      R2      F Pr(>F)  
## age                  1   0.2518 0.05403 1.1276  0.287  
## Cperfringens_VF      1   0.6597 0.14156 2.9542  0.018 *
## age:Cperfringens_VF  1   0.1756 0.03768 0.7863  0.643  
## Residual            16   3.5732 0.76672                
## Total               19   4.6603 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(pfam_jac ~  Cperfringens_VF + location, data = pfam_df)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Permutation: free
## Number of permutations: 999

## adonis2(formula = pfam_jac ~ Cperfringens_VF + location, data = pfam_df)
##                 Df SumOfSqs      R2      F Pr(>F)  
## Cperfringens_VF  1   0.6200 0.13303 2.6391  0.023 *
## location         3   0.5167 0.11087 0.7332  0.771  
## Residual        15   3.5237 0.75610                
## Total           19   4.6603 1.00000                
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

beta <- betadisper(pfam_jac, pfam_df$Cperfringens_VF)
permutest(beta)

## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999

## Response: Distances
##           Df  Sum Sq  Mean Sq     F N.Perm Pr(>F)  
## Groups     1 0.09820 0.098201 3.825    999  0.076 .
## Residuals 18 0.46212 0.025673                      
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
```
Protein families are significantly different between groups with and without _C. perfringens_ virulence factors. This remains true when first removing age effects, or when setting the model with age and _C. perfringens_ as interactive terms. PERMDISP was not significant for those groups. No evidence of protein family variation between age classes, males and females, or location. Now ordinate protein families based on jaccard distance. 
```{r eval=FALSE}
pcoa <- ordinate(
  physeq = pfamme_sub, 
  method = "PCoA", 
  distance = "jaccard"
)
```
Plot PCoA using ordination, and use the significance value from the first PERMANOVA. 
```{r eval=FALSE}
cperf_pfam <- expression(italic(paste("Protein family clustering (P=0.02)")))

pfam_pcoa <- plot_ordination(
  physeq = pfamme_sub,
  ordination = pcoa,
  color = "Cperfringens_VF", 
  title = cperf_pfam
) + 
  scale_color_manual(name=NULL,
                     values=c("black", "red"),
                     breaks=c("N", "Y"),
                     labels=c("No VFs", "VFs")
  ) +
  geom_point(aes(color = Cperfringens_VF), alpha = 1, size = 2.5) +
  stat_ellipse() + 
  coord_cartesian() + 
  theme_light()
pfam_pcoa <- pfam_pcoa + theme(text = element_text(family = "serif"))
pfam_pcoa
```
Now return to 16S data to see if this variation persists outside of the shotgun sequencing data. Load the 16S data and limit the phyloseq object to samples with corresponding WGS data. 
```{r eval=FALSE}
mothur <- import_mothur(mothur_shared_file = "~/Research/Data/16S_Penguins_2018/R_penguins/GAPE2018.shared", mothur_constaxonomy_file = "~/Research/Data/16S_Penguins_2018/R_penguins/GAPE2018.taxonomy")
metadata_16s <- read_excel(path="~/Research/Data/16S_Penguins_2018/R_penguins/GAPE_metadata.xlsx")
meta16s <- sample_data(metadata_16s) 
rownames(meta16s) <-meta16s$id 
gape <- merge_phyloseq(meta16s, mothur)
colnames(tax_table(gape)) <- c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus" )
sub <- subset_taxa(gape, Phylum != "Bacteria_unclassified")
sub <- rarefy_even_depth(sub, sample.size = 11295,
                         rngseed = TRUE, replace = TRUE, trimOTUs = TRUE, verbose = TRUE)

wgs16s = subset_samples(sub, also_wgs=="Y")
wgs16s
```
Use PERMANOVA to test significance between groups while controlling for location. Include age in the model. Verify PERMDISP. 
```{r eval=FALSE}
library(vegan)
wgs16s_jac <- phyloseq::distance(wgs16s, method = "jaccard")
wgs16s_df <- data.frame(sample_data(wgs16s))

perm <- how(nperm = 999)
setBlocks(perm) <- with(wgs16s_df, location)


adonis2(wgs16s_jac ~ age + Cperfringens_VF, data = wgs16s_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(wgs16s_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = wgs16s_jac ~ age + Cperfringens_VF, data = wgs16s_df, permutations = perm)
##                 Df SumOfSqs      R2      F Pr(>F)    
## age              1   0.4634 0.09733 2.2710  0.017 *  
## Cperfringens_VF  1   0.8289 0.17409 4.0619  0.001 ***
## Residual        17   3.4689 0.72858                  
## Total           19   4.7612 1.00000                  
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(wgs16s_jac ~ Cperfringens_VF, data = wgs16s_df, permutations=perm)

## Permutation test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(wgs16s_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = wgs16s_jac ~ Cperfringens_VF, data = wgs16s_df, permutations = perm)
##                 Df SumOfSqs     R2      F Pr(>F)   
## Cperfringens_VF  1   0.8375 0.1759 3.8419  0.002 **
## Residual        18   3.9237 0.8241                 
## Total           19   4.7612 1.0000                 
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

adonis2(wgs16s_jac ~ age*Cperfringens_VF, data = wgs16s_df, permutations=perm)
##  test for adonis under reduced model
## Terms added sequentially (first to last)
## Blocks:  with(wgs16s_df, location) 
## Permutation: free
## Number of permutations: 999

## adonis2(formula = wgs16s_jac ~ age * Cperfringens_VF, data = wgs16s_df, permutations = perm)
##                     Df SumOfSqs      R2      F Pr(>F)   
## age                  1   0.4634 0.09733 2.2552  0.018 * 
## Cperfringens_VF      1   0.8289 0.17409 4.0337  0.002 **
## age:Cperfringens_VF  1   0.1812 0.03805 0.8817  0.525   
## Residual            16   3.2877 0.69053                 
## Total               19   4.7612 1.00000                 
## ---
## Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

beta <- betadisper(wgs16s_jac, wgs16s_df$Cperfringens_VF)
permutest(beta)

## Permutation test for homogeneity of multivariate dispersions
## Permutation: free
## Number of permutations: 999

## Response: Distances
##           Df   Sum Sq   Mean Sq      F N.Perm Pr(>F)
## Groups     1 0.007252 0.0072517 1.2646    999   0.27
## Residuals 18 0.103215 0.0057342 
```
Based on 16S data, groups with and without _C. perfringens_ are significantly different through a PERMANOVA controlled for location. This remains true when first removing age effects, or when setting the model with age and _C. perfringens_ as interactive terms. PERMDISP is not significant for these groups. Now ordinate 16S data using jaccard distance. 
```{r eval=FALSE}
wgs_pcoa <- ordinate(
  physeq = wgs16s, 
  method = "PCoA", 
  distance = "jaccard"
)
```
Plot PCoA of groups with and without _C. perfringens_ virulence factors using significance value from the first PERMANOVA, which controls for location and removes the effects of age variation. 
```{r eval=FALSE}
cperf16s <- expression(italic(paste("OTU clustering (P=0.001)")))

tax16s <- plot_ordination(
  physeq = wgs16s,
  ordination = wgs_pcoa,
  color = "Cperfringens_VF",
  title = cperf16s
) + 
  geom_point(aes(color = Cperfringens_VF), alpha = 1, size = 2.5) +
  scale_color_manual(name=NULL, 
                     values=c("black", "red"), 
                     breaks=c("N", "Y"), 
                     labels=c("No VFs", "VFs")) +
  stat_ellipse() + 
  coord_cartesian() + 
  theme_light()
tax16s <- tax16s + theme(text = element_text(family = "serif"))
tax16s
```
Combine PCoAs from 16S, metabolic pathway, and protein family datasets, as well as the significantly different metabolic groups. 
```{r eval=FALSE}
row <- plot_grid(tax16s, pfam_pcoa, ncol = 2, labels = "AUTO")
row2 <- plot_grid(metab_pcoa, metab_groups, ncol = 2, labels = c('C', 'D'))
plot_grid(NULL, row, row2, nrow = 3, rel_heights = c(0.01, 1, 1)) + 
  theme_bw()
```

```{r echo=FALSE, eval=FALSE }
ggsave("cperf_cowplot.png", width = 9, height = 6.5, dpi = 300)
```

```{r echo=FALSE, out.width = '100%'}
knitr::include_graphics("cperf_cowplot.png")
```
Finally, check alpha diversity between groups with and without virulence factors. 
```{r eval=FALSE}
kruskal.test(observed ~ Cperfringens_VF, data = metadata_16s)
## Kruskal-Wallis rank sum test
## data:  observed by Cperfringens_VF
## Kruskal-Wallis chi-squared = 2.0543, df = 2, p-value = 0.358

kruskal.test(shannon ~ Cperfringens_VF, data = metadata_16s)
## Kruskal-Wallis rank sum test
## data:  shannon by Cperfringens_VF
## Kruskal-Wallis chi-squared = 1.238, df = 2, p-value = 0.5385

kruskal.test(simpson ~ Cperfringens_VF, data = metadata_16s)
## Kruskal-Wallis rank sum test
## data:  simpson by Cperfringens_VF
## Kruskal-Wallis chi-squared = 1.8025, df = 2, p-value = 0.4061
```
No variation apparent in alpha diversity between groups. 

## _References_
Some scripts were adapted from Phyloseq examples (https://joey711.github.io/phyloseq/index.html) and from examples on Riffomonas by Dr. Pat Schloss (https://riffomonas.org/). 
